packages/web/src/utils/config.ts

```diff
@@ -198,3 +200,34 @@ function getStatsConfig(options: BuildBuilderOptions): Stats.ToStringOptions {
     usedExports: !!options.verbose
   };
 }
+
+// This is shamelessly taken from CRA and modified for NX use
+// https://github.com/facebook/create-react-app/blob/4784997f0682e75eb32a897b4ffe34d735912e6c/packages/react-scripts/config/env.js#L71
+function getClientEnvironment(mode) {
+  // Grab NODE_ENV and NX_* environment variables and prepare them to be
+  // injected into the application via DefinePlugin in webpack configuration.
+  const NX_APP = /^NX_/i;
+
+  const raw = Object.keys(process.env)
+    .filter(key => NX_APP.test(key))
+    .reduce(
+      (env, key) => {
+        env[key] = process.env[key];
+        return env;
+      },
+      {
+        // Useful for determining whether we’re running in production mode.
+        NODE_ENV: process.env.NODE_ENV || mode
+      }
+    );
+
+  // Stringify all values so we can feed into webpack DefinePlugin
+  const stringified = {
+    'process.env': Object.keys(raw).reduce((env, key) => {
```

Comment:
Rather than `process.env` which will cause build errors, should we add a globally declared property like `nx` (and then `nx.env`)? 

The build fails with errors if we use `process.env` because when it's compiling, it's not node JS.

You _could_ of course declare a global `process` variable instead, but I wonder if that would be misleading?

@jaysoo What do you think?

Reply:
That's a good point @jdpearce.

Although would be nice for us to keep using the common/known `process.env` and instead find a fix for the global variable `process`. I will see what I can do about it.